ootPhotons for 80X Legacy reprocessing

kmcdermo opened the pull request:
This PR sits on top of PR #19205 (ootPhotons in miniAOD) with only one new commit for the legacy sequence.

I was able to get this to run locally and produce the ootPhoton collection from 80X AOD (2016G, singlephoton).  However, after rebasing, I cannot complete the runTheMatrix.py -l limited -i all because of the same problems with #19205, namely some PhaseII sequence breaking the workflow.

I had to touch a number of files to accommodate the differences between the new 92X interfaces and what is available in the 80X AOD.  A number of products have to be regenerated for this sequence to run (see the customization in the ootPhotonProducer_cff.py).  I tried following what was done in #19304 for customizing the configuration, as well as following along #19264 for minimally changing any code.  

EDIT:  I forgot to mention I built this on top of an IB: CMSSW_9_2_X_2017-06-21-2300, so as to include those two PRs listed above.

kmcdermo committed:
OOT photon sequence (RECO+PAT) for 2016 legacy 80X reminiAOD from AOD

cmsbuild commented:
A new Pull Request was created by  @kmcdermo (Kevin McDermott) for master.

It involves the following packages:

PhysicsTools/PatAlgos
RecoEcal/EgammaClusterProducers
RecoEgamma/EgammaIsolationAlgos
RecoEgamma/EgammaPhotonProducers
RecoEgamma/PhotonIdentification
RecoParticleFlow/PFClusterProducer


@perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please review it and eventually sign? Thanks.
@rappoccio, @gouskos, @TaiSakuma, @argiro, @seemasharmafnal, @mmarionncern, @imarches, @ahinzmann, @acaudron, @lgray, @jdolen, @ferencek, @Sam-Harper, @rafaellopesdesa, @nhanvtran, @gkasieczka, @schoef, @JyothsnaKomaragiri, @mverzett, @cbernet, @gpetruc, @mariadalfonso, @pvmulder, @bachtis this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

kmcdermo commented:
@slava77 these tests will fail for the same reason as #19205, namely some PhaseII tracking sequence.

cmsbuild commented:
+1
Tested at: a891cbea48ec16dbd75f22e2b64e41f5b34edf39
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/20843/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/4bfcf97d60143ae1a56e803de7148e9d3ab6787e
https://github.com/cms-sw/cmssw/commit/1c5bd71bbc32a44aa0f0a9f9a797aab45b556f3f
https://github.com/cms-sw/cmssw/commit/8730b28a4110dcf3324f0c8f02fe5e4991840f94
https://github.com/cms-sw/cmssw/commit/259c301a5949bee03aaab3e1cfa6694a50d2ab1e
https://github.com/cms-sw/cmssw/commit/4e50f8623ce65b3e8f2290271755366c5cbc0788
https://github.com/cms-sw/cmssw/commit/d0a212b197fd609cda9afb5c965c6daa0ba6b5b8
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/20843/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/20843/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/20843/summary.html

Comparison Summary:
 - You potentially added 3 lines to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 7 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1802916
 - DQMHistoTests: Total failures: 29459
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1773291
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

kmcdermo commented:
I made a simple validation of this sequence by re-processing some signal MC from AODSIM produced through the standard 80X chain, then switching to 92X for the PAT step using this era customization (GMSB, ctau = 3m).

Without any selection on the leading photon, this is the leading photon seed time for each photon collection produced in the PAT step (slimmedPatPhotons from gedPhotons and slimmedOOTPATPhotons from ootPhotons): 
![leadpho_time_nosel](https://user-images.githubusercontent.com/7645989/27530353-9a53fb34-5a58-11e7-9b3d-c1cb8bba537f.png)


However, as we have seen before, when we include some selection on the photon (namely, implementing the cut based VID: https://twiki.cern.ch/twiki/bin/view/CMS/CutBasedPhotonIdentificationRun2), the standard collection loses its late time photons, in which case our ootPhotons kick in: 
![leadpho_time_wsel](https://user-images.githubusercontent.com/7645989/27530356-9fcf2944-5a58-11e7-98a5-e9be87f7d8b0.png)


I would say this validates this sequence.

slava77 commented:
@kmcdermo 
this PR can now be advanced to the proper review step, since the #19205 has just been merged.
Please rebase and possibly apply already the updates mentioned for the ReducedEGProducer to continue

Thank you.

kmcdermo commented:
@slava77 

I guess rebase to the latest IB with 19205?  should I make a separate PR for the changes requested for the ReducedEGProducer or commit them here?  

For the sake of the backport, it would probably be easier with a separate PR

slava77 commented:
On 6/29/17 7:53 AM, Kevin McDermott wrote:
> @slava77 <https://github.com/slava77>
> 
> I guess rebase to the latest IB with 19205? should I make a separate PR
> for the changes requested for the ReducedEGProducer or commit them here?
> 
> For the sake of the backport, it would probably be easier with a separate PR

if after rebase this PR does not have to have changes in
ReducedEGProducer to handle legacy repro,
it should be OK to have fixes for ReducedEGProducer made in a separate PR.
If there is some coupling, that separate PR will delay this one further.

> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19404#issuecomment-311991751>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcboVEQz3r0nVFgQDTzYTiMhq16NMrks5sI7p4gaJpZM4OCQgL>.
>

kmcdermo committed:
remove dead assignment of index at the very end of producer

cmsbuild commented:
Pull request #19404 was updated. @perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please check and sign again.

kmcdermo commented:
@slava77 I rebased and fixed the dead assignment of index.

However, I am not so sure how to proceed on the range based loops and what the based way to this is.

For example: https://github.com/kmcdermo/cmssw/blob/f9d409d7928550bfa8892cd4f8c55747e511e6f4/RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc#L208-L211

could be converted to:
int index = 0;
```
std::vector<edm::Handle<edm::ValueMap<bool> > > photonIdHandles(photonIdTs_.size());
for (auto& photonIdT : photonIdTs)
    theEvent.getByToken(photonIdT,photonIdHandles[index++]);
  }
```
and so on for the rest of the other handles (and similar for L301-307).

----- 
for https://github.com/kmcdermo/cmssw/blob/f9d409d7928550bfa8892cd4f8c55747e511e6f4/RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc#L338-L365

I could convert the top of the loop to:
```
index = 0;
  for (auto& ootPhoton : *ootPhotonHandle) {
```
however, this gets messy then with updating index++, as it is needed for the PhotonRef immediately after a continue statement.

This is true for other places in the code as well (also including the regular photon loop, gsfElectrons, linking SCs: L418, etc.)

This strategy could be applied to L673, L689.  

Let me know.

slava77 commented:
keeping a running index in parallel seems to be practical. In most cases it is needed only once.
So, an ```index++``` in place would do.
Overall the savings is rather minimal: to avoid in-loop handle dereferencing and position extraction.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: f9d409d7928550bfa8892cd4f8c55747e511e6f4
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21014/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/b493ac04b2a3beaba92e74f17e4e1b80c57b3392
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21014/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21014/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21014/summary.html

Comparison Summary:
 - You potentially added 3 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1756063
 - DQMHistoTests: Total failures: 97
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1755800
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

kmcdermo committed:
switch to range based loops everywhere in ReducedEGProducer

kmcdermo commented:
@slava77 
I made the changes as suggested.  I was not able to run locally, as cmsDriver crashed with "ImportError: No module named Applications"...

Trying to reuse an old config for testing this branch also failed to execute with: 
"No module named StandardSequences.Eras"

davidlange6 commented:
@cmsbuild please test

that sounds like a sign that your local area is in an unhappy state. (perhaps deleting some directories?)

> On Jun 30, 2017, at 12:00 PM, Kevin McDermott <notifications@github.com> wrote:
> 
> @slava77
> I made the changes as suggested. I was not able to run locally, as cmsDriver crashed with "ImportError: No module named Applications"...
> 
> Trying to reuse an old config for testing this branch also failed to execute with:
> "No module named StandardSequences.Eras"
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

cmsbuild commented:
The tests are being triggered in jenkins.

kmcdermo commented:
@davidlange6 
Most probably. I made fresh local area and everything worked locally (able to see ootPhotons in miniAOD from some signal samples).

cmsbuild commented:
+1
Tested at: 35c56c09c91da0fc81b223173b8ae19deec34d49
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21021/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/97467bb9b4552da8afe894507ed4ca9404681e12
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21021/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21021/git-merge-result

cmsbuild commented:
Comparison job queued.

slava77 commented:
@cmsbuild please test workflow 136.7611,1325.5

cmsbuild commented:
The tests are being triggered in jenkins.

slava77 commented on PhysicsTools/PatAlgos/python/producersLayer1/ootPhotonProducer_cff.py:
```diff
@@ -49,3 +49,44 @@
+    particleFlowSuperClusterOOTECAL,
+    ootPhotonCore,
+    ootPhotons,
+    ootPhotonEcalPFClusterIsolationProducer
```
if I'm not mistaken, this is a re-definition of the ootPhotonSequence as a taks.
This should be moved to RecoEgamma/EgammaPhotonProducers/python/ootPhotonSequence_cff.py
and whatever upstream files from that ones. In there please define a task and then initialize the necessary cms.Sequence 

Eventually, updates in this file should look like [modulo corrections for typos I may have below]
```python
from RecoEgamma.EgammaPhotonProducers.ootPhotonSequence_cff import ootPhotonTask

+from Configuration.Eras.Modifier_run2_miniAOD_80XLegacy_cff import run2_miniAOD_80XLegacy
run2_miniAOD_80XLegacy.toReplaceWith(makePatOOTPhotonsTask, cms.Task(makePatOOTPhotons.copy(),ootPhotonTask) )

run2_miniAOD_80XLegacy.toModify(patOOTPhotons, hcalPFClusterIsoMap = "")
```

kmcdermo replied:
@slava77 

I think I made the changes as you suggested.  However, there are a few modules that are needed in this legacy task that already included in the standard RECO: 

- CaloTowerConstituentsMapBuilder
- essourceEcalSev 
- ecalSeverityLevel
- particleFlowRecHitPS
- particleFlowClusterPS
- ootPhotonEcalPFClusterIsolationProducer

I added these (and there imports) to this file.  Please review the changes to this file and RecoEgamma/EgammaPhotonProducers/python/ootPhotonSequence_cff.py and let me know if it needs more modification.

slava77 commented on RecoEgamma/EgammaIsolationAlgos/interface/EgammaTowerIsolation.h:
```diff
@@ -220,7 +220,9 @@ class EgammaTowerIsolation {
-  
+
+
+  void setTLS(const CaloTowerCollection* towers);  
```
If this method is not needed anywhere other than in the constructor, just fold it back to the constructor.

kmcdermo replied:
done.

slava77 commented on RecoEgamma/EgammaPhotonProducers/python/reducedEgamma_cfi.py:
```diff
@@ -74,3 +74,10 @@
+run2_miniAOD_80XLegacy.toModify(
+    reducedEgamma, 
+    ootPhotonPFClusterIsoSources = cms.VInputTag(cms.InputTag("ootPhotonEcalPFClusterIsolationProducer")),
+    ootPhotonPFClusterIsoOutput = cms.vstring("ootPhoEcalPFClusIso")
```
can we avoid type specifications here?
e.g. ```ootPhotonPFClusterIsoSources =  [  "ootPhotonEcalPFClusterIsolationProducer" ] ``` ?

kmcdermo replied:
I wasn't sure how to do this, so I made with the type specification -- thanks for the example!  done.

slava77 commented on RecoEgamma/EgammaPhotonProducers/src/GEDPhotonProducer.cc:
```diff
@@ -503,17 +506,21 @@ void GEDPhotonProducer::fillPhotonCollection(edm::Event& evt,
-    double HoE1=towerIso1.getTowerESum(&(*scRef))/scRef->energy();
-    double HoE2=towerIso2.getTowerESum(&(*scRef))/scRef->energy(); 
+    double HoE1 = (hcalTowersColl != nullptr ? towerIso1.getTowerESum(&(*scRef))/scRef->energy() : 0.f);
+    double HoE2 = (hcalTowersColl != nullptr ? towerIso2.getTowerESum(&(*scRef))/scRef->energy() : 0.f); 
```
for all of this : why not simply
```
double HoE1,HoE2;
HoE1=HoE2=0;
if (not hcalTowers_.isUninitialized()){
//all old implementation
}
```
this way there is also no need to modify EgammaTowerIsolation

kmcdermo replied:
done.

slava77 commented on RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc:
```diff
@@ -204,31 +204,38 @@ void ReducedEGProducer::produce(edm::Event& theEvent, const edm::EventSetup& the
   std::vector<edm::Handle<edm::ValueMap<float> > > ootPhotonPFClusterIsoHandles(ootPhotonPFClusterIsoTs_.size());
-  for (unsigned int itok=0; itok<ootPhotonPFClusterIsoTs_.size(); ++itok) {
-    theEvent.getByToken(ootPhotonPFClusterIsoTs_[itok],ootPhotonPFClusterIsoHandles[itok]);
+  for (auto& ootPhotonPFClusterIsoT : ootPhotonPFClusterIsoTs_) {
```
```auto const& ``` ?

kmcdermo replied:
done (and above and below).

slava77 commented on RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc:
```diff
@@ -284,26 +291,28 @@ void ReducedEGProducer::produce(edm::Event& theEvent, const edm::EventSetup& the
   //loop over photons and fill maps
-  for (unsigned int ipho=0; ipho<photonHandle->size(); ++ipho) {
-    const reco::Photon &photon = (*photonHandle)[ipho];
+  for (const auto& photon : *photonHandle) {
```
it make more sense to me to set index=0 just above instead of 50 lines above
.. also, if you set "index = -1" outside of the loop, the first line in the loop can be "index++"
and there is no need to keep track of all possible places where the increment can be missed.

kmcdermo replied:
> it make more sense to me to set index=0 just above instead of 50 lines above

done.

> .. also, if you set "index = -1" outside of the loop, the first line in the loop can be "index++"
and there is no need to keep track of all possible places where the increment can be missed.

done.

slava77 commented on RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc:
```diff
@@ -363,26 +374,29 @@ void ReducedEGProducer::produce(edm::Event& theEvent, const edm::EventSetup& the
+    if (!keep) {index++; continue;}
     
-    reco::GsfElectronRef gsfElectronref(gsfElectronHandle,iele);
+    reco::GsfElectronRef gsfElectronref(gsfElectronHandle,index++);
```
"index = -1" will work here as well

kmcdermo replied:
done (and everywhere else).

slava77 commented on RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc:
```diff
@@ -459,6 +473,7 @@ void ReducedEGProducer::produce(edm::Event& theEvent, const edm::EventSetup& the
     // link calo clusters
     linkCaloClusters(ootSuperCluster, *ootEbeeClusters, ootEbeeClusterMap, rechitMap, barrelHitHandle, endcapHitHandle, caloTopology, *ootEsClusters, ootEsClusterMap);
   }
+  index = 0;
```
as above: reset it only right before the loop where it's needed instead of right after the last loop where it was used

kmcdermo replied:
done (and everywhere else).

slava77 commented on RecoEgamma/PhotonIdentification/src/PhotonIsolationCalculator.cc:
```diff
@@ -661,19 +664,19 @@ double PhotonIsolationCalculator::calculateHcalTowerIso(const reco::Photon* phot
                               eMin,depth,
                               toww);
-  hcalIsol = phoIso.getTowerEtSum(photon);
+  hcalIsol = (toww != nullptr ? phoIso.getTowerEtSum(photon) : 0.);
```
as above: better move the computation behind the ```if (not hcalCollection_.isUninitialized()){``` to avoid unnecesary initialization/instantiation of the phoIso

kmcdermo replied:
done.

slava77 commented on RecoParticleFlow/PFClusterProducer/python/particleFlowRecHitOOTECAL_cff.py:
```diff
@@ -1,6 +1,34 @@
+run2_miniAOD_80XLegacy.toModify(
+    particleFlowRecHitOOTECAL.producers[0].qualityTests[0], 
+    name = "PFRecHitQTestThreshold",
+    threshold = cms.double(0.08),
```
is this change to qualityTests actually needed? (Run2_2016 to be used together with run2_miniAOD_80XLegacy is supposed to already have the right thresholds)

kmcdermo replied:
Ah, I didn't make this change with these commits. Where can I see the era-dependent (i.e. for Run2_2016) thresholds?  It was not present in: RecoParticleFlow/PFClusterProducer/python/particleFlowRecHitECAL_cfi.py

slava77 replied:
in RecoParticleFlow/PFClusterProducer/python/particleFlowRecHitECAL_cfi.py
```python
                  name = cms.string("PFRecHitQTestECALMultiThreshold"),
                  thresholds = particle_flow_zero_suppression_ECAL.thresholds
```
where particle_flow_zero_suppression_ECAL is from RecoParticleFlow/PFClusterProducer/python/particleFlowZeroSuppressionECAL_cff.py and its default value for 2016 and earlier is the thresholds that you need (they are the same as here: EB at 80 MeV and EE at 300 MeV

Hence, these modifications are not necessary in this \*OOT\* file.

kmcdermo replied:
Right, right.  I forgot, thanks!  I will make this change.

cmsbuild commented:
+1
Tested at: 35c56c09c91da0fc81b223173b8ae19deec34d49
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21038/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/97467bb9b4552da8afe894507ed4ca9404681e12
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21038/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21038/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21038/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1756063
 - DQMHistoTests: Total failures: 15020
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1740877
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

slava77 commented:
The reminiAOD workflows are working now 136.7611, 1325.5 (they are failing in the IBs)

slava77 commented:
@kmcdermo 
after some more thought, I now realize that a backport will be needed for this as well to 92X.
It shall better be done combined with the regular OOT miniAOD,
because the backport of the regular OOT miniAOD alone will break the legacy reminiAOD workflows
either for those who want to test it in 92X or for the relval matrix tests introduced in (still not merged) #19450

kmcdermo commented:
> after some more thought, I now realize that a backport will be needed for this as well to 92X.
It shall better be done combined with the regular OOT miniAOD,
because the backport of the regular OOT miniAOD alone will break the legacy reminiAOD workflows
either for those who want to test it in 92X or for the relval matrix tests introduced in (still not merged) 
@slava77 
Ok, sorry for the delay in response.  I will get to work on the corrections as suggested (I have a few questions).

As for the backport of this PR with the regular miniAOD ootPhotons, I guess I should just cherry-pick commits from this branch onto the regular ootPhoton miniAOD branch PR (that way it can all be merged as one)?

kmcdermo committed:
undo changes to EgammaTowerIsolation, minimize impact of missing hcaltowers on GEDPhotonProducer and PhotonIsolationProducer by hiding calculations behind if blocks as suggested

kmcdermo committed:
make changes as suggested for ReducedEGProducer: move index initialization to before loops, const auto where possible, use index=-1 in EGM object loops to avoid keeping track of incrementing index

kmcdermo committed:
make changes to configs as requested: remove type spec. in reducedEgamma_cfi, make ootPhotons a task in ootPhotonSequence_cff, import task and modify accordingly for legacy task in ootPhotonProducer_cff

cmsbuild commented:
Pull request #19404 was updated. @perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please check and sign again.

kmcdermo commented:
@slava77 
Okay, I made all the changes as requested.  Tested locally on SinglePhoton AOD 2016G data file, worked as expected.

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 74fe1f75a8209bd65f282f3346e8f285f81d6f0a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21065/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5387f33035f82196d51968058b52bc670f79533c
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21065/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21065/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21065/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1756063
 - DQMHistoTests: Total failures: 14918
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 1740978
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

slava77 commented on PhysicsTools/PatAlgos/python/producersLayer1/ootPhotonProducer_cff.py:
```diff
@@ -49,3 +49,27 @@
+                                     ecalSeverityLevel,
+                                     particleFlowRecHitPS,
+                                     particleFlowClusterPS,
+                                     ootPhotonTask,
```
if the ES and \*PS modules are needed to make ootPhotonTask to work, they better be added to ootPhotonTask

kmcdermo replied:
well, they are not needed in the standard reco, only in this special case.

or rather, they are already produced in the standard reco chain, and therefore available, whereas here they have to be reproduced.

kmcdermo replied:
let me know if I should move these things to the ootPhotonTask

slava77 replied:
It's a fair argument that ootPhotonTask needs these extra modules only in the legacy reminiAOD setup.
The modifier can be applied to ootPhotonTask where it's defined. This PAT producer file isn't supposed to know what else is needed logically upstream of making ootPhoton

kmcdermo replied:
Okay, fair enough, I will make this change.

kmcdermo committed:
move changes to ootPhotonTask to ootPhotonSequence_cff, remove superfluous modifiers for legacy repro on recHits

cmsbuild commented:
Pull request #19404 was updated. @perrotta, @cmsbuild, @slava77, @monttj, @davidlange6 can you please check and sign again.

kmcdermo commented:
okay, I have made the changes as recommended, and the tests locally worked.

slava77 commented:
@cmsbuild please test workflow 136.7611,1325.5

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 73da68715b884dc85c81db3fdb9ea3f10be8098c
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21100/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/b1580cdb9b247f8d4a93ce44ec73c8c8213a595e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21100/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21100/git-merge-result

cmsbuild commented:
Comparison job queued.

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/19404	 	73da687
- added modifications for run2_miniAOD_80XLegacy to be able to run OOT photons in miniAOD from 2016 AOD
- jenkins tests pass including now running reminiAOD workflows 136.7611,1325.5
- local tests with 136.7611,1325.5 show roughly expected behavior (the baseline used for tests is CMSSW_9_3_X_2017-07-02-2300 with #19205 reverted): OOT photon modules run and the final miniAOD contains slimmedOOTPhotons reco::Photon collection and also additional relevant reducedEgamma clusters and hits. Changes are very similar to what was observed in signoff of #19205 
    - 2K event run in 136.7611 brings 127 slimmedOOTPhotons, 185 basic clusters and an increase of 1.3% in the size of the reducedEgamma:reducedEBRecHits. The total added products are about 400 bytes per event.
    - CPU cost per event is somewhat negligible (about 30 ms/event compared to 1.8 s/event in this reminiAOD job in full)

davidlange6 commented:
merge

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19404/21100/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1756063
 - DQMHistoTests: Total failures: 29449
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 1726447
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files
